.github/workflows/README.md

```diff
@@ -0,0 +1,57 @@
+# Workflow Docs
+
+## Tmate action
+
+The following is a rundown of the tmate action used in most of the workflows. Its structure looks something like this:
+
+```github-action
+  - name: Setup tmate session
+    if: ${{ failure() }}
+    uses: mxschmitt/action-tmate@v3.18
+    timeout-minutes: 15
+    with:
+        detached: false
+        limit-access-to-actor: true
+```
+
+While it may seem obvious to some, it is important to note that the workflow will not complete until the tmate action step completes.
+Since we have concurrency set on most of our workflows, this means that if you push another run of the workflow, you must first close your SSH session.
+More information on this is available in the following section [When / Why does the action step close](./README.md#when--why-does-the-action-step-close).
+It is for this reason that this may not be useful in every situation.
+
+### When / Why does the action step close?
+
+This action will wait for one of two cases, the first of which is connection close. The SSH session only supports a single connection,
+if you ssh and close the connection the action step will close and the workflow will proceed, even if you have not finished the `timeout-minutes` window.
+
+This answers the question posed in the previous section, if you have the session enabled, and want to fail this run to get to the next run of your workflow,
+simply run the ssh command provided in the logs of the action step (ex: `ssh <provided_ip>@nyc1.tmate.io`), and then close your session manually.
+This will exit the tmate action step, and allow the workflow to continue, regardless of if it runs in `detached: true` or `detached: false` mode.
+
+Note also that as it only supports a single connection, only one person can ssh to the tmate sessions, others will be rejected.
+The second condition is that the `timeout-minutes` elapse, in which case the action will boot you out of ssh, the session will close and the worfklow will continue.
+
+### Configurations
+
+The key values are `timeout-minutes`, `detached` and `limit-access-to-actor`.
+
+#### Detached mode
+
+If the action step is ran with `detached: true`, it will proceed to the next action steps unhindered.
+If the workflow finishes before the `timeout-minutes` has elapsed, it will pop open a new action step at the end of the workflow to wait for and cleanup the tmate action.
+If the step is instead ran with `detached: false` the workflow will not proceed until the step closes.
+
+#### Limit access to actor
+
+With `limit-access-to-actor` set to `true`, the action look who created the PR, and grab the public SSH keys stored in their Github account.
+It will reject connections from any SSH private key that does not match the public key listed in the Github account.
+This is recommended, as it prevents others from abusing your runners, but may be dissabled to allow a teamate to ssh instead.
```

Comment:
This seems like a really bad idea to allow an arbitrary PR author to ssh into a CI runner. Is that what this does?

Reply:
With `limit-access-to-actor` set to `true`, it will check the the public ssh keys saved to the account of the person who opened the PR. It will only accept ssh connections from those SSH private keys that are paired to the public keys.

Reply:
Right -- so bad actor A can open a PR and then ssh in to use it to mine MeowCoin or whatever

or worse, bad actor A gets to ssh in to build and push stuff to our image registry.

Reply:
Got it, closing the issues then. Although I think this is more of a CI permissions issue ... because someone could still just include this action in any pr they run and achieve the same result. Regardless I will close this up now, if we want we can always reopen it.

Reply:
That's true. I believe we have it set so at least the first time someone submits a PR an approval is required for CI to run. That's a pretty low bar to get past, though. We've got to change the setting to always requiring approval for outside collaborators.

<img width="902" alt="image" src="https://github.com/instructlab/instructlab-bot/assets/309258/43f1a542-8317-479a-8f82-b6b456f26f91">